File: backend/src/main/java/com/woowacourse/momo/category/domain/Category.java
package com.woowacourse.momo.category.domain;

import java.util.Arrays;

import lombok.Getter;
import lombok.RequiredArgsConstructor;

import com.woowacourse.momo.category.exception.CategoryErrorCode;
import com.woowacourse.momo.category.exception.CategoryException;

@Getter
@RequiredArgsConstructor
public enum Category {

    STUDY(1, "스터디", "thumbnail_study.jpg", "icon_study.svg"),
    MOCO(2, "모각코", "thumbnail_moco.jpg", "icon_moco.svg"),
    EAT(3, "식사", "thumbnail_eat.jpg", "icon_eat.svg"),
    CAFE(4, "카페", "thumbnail_cafe.jpg", "icon_cafe.svg"),
    DRINK(5, "술", "thumbnail_drink.jpg", "icon_drink.svg"),
    HEALTH(6, "운동", "thumbnail_health.jpg", "icon_health.svg"),
    GAME(7, "게임", "thumbnail_game.jpg", "icon_game.svg"),
    TRAVEL(8, "여행", "thumbnail_travel.jpg", "icon_travel.svg"),
    CULTURE(9, "문화생활", "thumbnail_culture.jpg", "icon_culture.svg"),
    ETC(10, "기타", "thumbnail_etc.jpg", "icon_etc.svg");

    private final long id;
    private final String name;
    private final String defaultImageName;
    private final String iconName;

    public static Category from(long id) {
        return Arrays.stream(values())
                .filter(category -> category.id == id)
                .findFirst()
                .orElseThrow(() -> new CategoryException(CategoryErrorCode.CATEGORY_NOT_EXIST));
    }

    public boolean isDefaultImage(String imageName) {
        return defaultImageName.equals(imageName);
    }
}

File: frontend/src/mocks/handlers/category.ts
import { rest } from 'msw';

import { baseURL } from 'apis/axios';
import { API_PATH } from 'constants/path';

const apiURL = `${baseURL}${API_PATH.CATEGORY}`;

export const categoryHandler = [
  // 카테고리 정보 조회
  rest.get(apiURL, (req, res, ctx) => {
    return res(
      ctx.status(200),
      ctx.json([
        {
          id: 1,
          name: '스터디',
          imageUrl: 'https://image.moyeora.site/category/icon_study.svg',
        },
        {
          id: 2,
          name: '모각코',
          imageUrl: 'https://image.moyeora.site/category/icon_moco.svg',
        },
        {
          id: 3,
          name: '식사',
          imageUrl: 'https://image.moyeora.site/category/icon_eat.svg',
        },
        {
          id: 4,
          name: '카페',
          imageUrl: 'https://image.moyeora.site/category/icon_cafe.svg',
        },
        {
          id: 5,
          name: '술',
          imageUrl: 'https://image.moyeora.site/category/icon_drink.svg',
        },
        {
          id: 6,
          name: '운동',
          imageUrl: 'https://image.moyeora.site/category/icon_health.svg',
        },
        {
          id: 7,
          name: '게임',
          imageUrl: 'https://image.moyeora.site/category/icon_game.svg',
        },
        {
          id: 8,
          name: '여행',
          imageUrl: 'https://image.moyeora.site/category/icon_travel.svg',
        },
        {
          id: 9,
          name: '문화생활',
          imageUrl: 'https://image.moyeora.site/category/icon_culture.svg',
        },
        {
          id: 10,
          name: '기타',
          imageUrl: 'https://image.moyeora.site/category/icon_etc.svg',
        },
      ]),
    );
  }),
];
